core/shared/src/main/scala/zio/prelude/ZNonEmptySet.scala

```diff
@@ -87,6 +87,9 @@ final class ZNonEmptySet[+A, +B] private (private val zset: ZSet[A, B]) { self =
   )(implicit ev1: Commutative[Sum[B1]], ev2: Commutative[Prod[B1]]): ZNonEmptySet[C, B1] =
     new ZNonEmptySet[C, B1](zset.flatMap(f(_).toZSet))
 
+  /** Returns an element */
+  def head: A = zset.toMap.head._1
```

Comment:
This is potentially a little weird in that the `head` could be an element with a measure that is negative or not integral. Somewhat beyond the scope of this PR but I think the concept of what it means to be `NonEmpty` is a little unclear. At the least that it doesn't have any elements with measures that are equal to an identity value? Not sure that is enforced right now.